alibuild_helpers/build.py

```diff
@@ -344,33 +344,30 @@ def doBuild(args, parser):
                   "  git pull --rebase\n",
                   pwd=os.getcwd(), star=star()))
 
-  # Prefetch git heads in parallel
-  debug("Prefetching git heads in parallel:")
-
-  fetchers = list(filter(lambda p: "source" in specs[p], buildOrder))
-  poolConfig = {"processes": min(max(len(fetchers), 1), 16)}
-  pool = ThreadPool(**poolConfig)
-  debug("Created thread pool with config: %s" % poolConfig)
-
-  git_commands = {} # store futures
-  for p in fetchers:
-    # Replace source with local one if we are in development mode.
+  # Clone/update repos
+  reposWithSource = list(filter(lambda p: "source" in specs[p], buildOrder))
+  if args.fetchRepos:
+    # Force update of reference repos
+    for p in reposWithSource:
+      updateReferenceRepos(args.referenceSources, p, specs[p])
+  else:
+    # Only create reference if it does not exist
+    for p in reposWithSource:
+      specs[p]["reference"] = "%s/%s" % (abspath(args.referenceSources), p.lower())
+      if not exists(specs[p]["reference"]):
+        updateReferenceRepos(args.referenceSources, p, specs[p])
+
+  # Retrieve git heads
+  git_head_fetchers = list(filter(lambda p: "reference" in specs[p], buildOrder))
```

Comment:
Here too we can have all in one line:

```python
for p in list(filter(lambda p: "reference" in specs[p], buildOrder)):
  ...
```

Reply:
Would it not be more idiomatic to use a list comprehension?
e.g.
```for p in [p for p in buildOrder if "reference" in specs[p]]:```

Reply:
Even better, yes, I fully agree.